### PR TITLE
[WIP] Explain "tricky" code related to orders

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Add topic where all "tricky code" related to orders should be explained
| Fixed ticket? | Fixes #{issue number here} if there is a related issue

### CODE REVIEW

File: src/content/1.7/development/behind-the-scenes/orders/_index.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+refactoring at this stage). Some of those methods, especially the ones related to cart modifications, are closely
+coupled with the Context. However, the context became inconsistent, because some of context related operations were
+hardcoded in controllers or other classes which are not used anymore. To make sure that the context contains what we
+expect before calling certain methods, we had introduced the
+the [ContextStateManager](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/src/Adapter/ContextStateManager.php).
+This service allows to modify the context and revert ir back to the original state. For example, it is possible to set
+the context `Cart` or a `Customer` to certain values, then perform some operations that depends on those values through a
+context and then revert it back to avoid unexpected behavior in further code.
+
+### Specific price for product in cart
```
Comment (zuk3975) [inline src/content/1.7/development/behind-the-scenes/orders/_index.md:25]: I think this part could go to page-reference/back-office/order/create

File: src/content/1.7/development/behind-the-scenes/orders/_index.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+In BO `Sell -> Orders -> Orders -> Add new order`, user can modify the price for selected product in cart. Behind the
+scenes, to achieve this, a
+new [SpecificPrice](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/classes/SpecificPrice.php) is created and
+assigned to that cart and that product.
+See [CartController::editProductPriceAction](https://github.com/PrestaShop/PrestaShop/blob/1.7.8.x/src/PrestaShopBundle/Controller/Admin/Sell/Order/CartController.php)
+.
+
+### Free shipping cart rule in cart
```
Comment (zuk3975) [inline src/content/1.7/development/behind-the-scenes/orders/_index.md:34]: same for this. Could be mentioned in order create page reference when its done.

File: src/content/1.7/development/behind-the-scenes/orders/_index.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+weight: 42
+
+chapter: true
+---
+
+# Orders related code specifics
+
+Explains some "not so obvious" technical parts related to orders domain.
+
+### Context state manager
```
Comment (zuk3975) [inline src/content/1.7/development/behind-the-scenes/orders/_index.md:13]: if all bellow items goes to order create page, then maybe we could find a place for context state manager somewhere more "general". and remove this "behind-the-scenes/orders" at all?

File: src/content/1.7/development/behind-the-scenes/orders/_index.md (hunk 1 above)
Comment (zuk3975) [reply src/content/1.7/development/behind-the-scenes/orders/_index.md:25]: moved out to https://github.com/PrestaShop/docs/pull/981

File: src/content/1.7/development/behind-the-scenes/orders/_index.md (hunk 2 above)
Comment (zuk3975) [reply src/content/1.7/development/behind-the-scenes/orders/_index.md:34]: moved out to https://github.com/PrestaShop/docs/pull/981

**Comment by zuk3975** [comment]
closing. 
Only context state manager need to be documented somewhere. Other parts were moved out

**Comment by matks** [comment]
🎉

**Pull request closed without merging**